### ISSUE TITLE
RulesEngine.check() does not throw error 

### ISSUE THREAD

**Issue by rahamathulla-ai** [j-easy/easy-rules#307]
RulesEngine.check() does not throw error if the then expression is failing. but the error is thrown when firing the rules.
is there any way we can check the syntax errors before firing the rules?

**Comment by benas** [comment]
> RulesEngine.check() does not throw error if the then expression is failing. but the error is thrown when firing the rules.

The Javadoc of `RulesEngine#check` says:

```
Check rules without firing them.
```

I think this is clear enough to imply that rules will not be fired (action methods will no be called, only the evaluation methods). So you shouldn't be expecting this method to throw errors in the then expression.

> is there any way we can check the syntax errors before firing the rules?

No, not at runtime (unless you are ready to pre-fire rules on dummy facts). I believe this shouldn't wait until runtime to be checked, rules must be unit-tested beforehand.
